risiko.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <cstdlib>
+#include <string>
+using namespace std;
+
+void switchSort(int* , int);
+
+class Player{
+    private:
+        string role;
+
+    public:
+
+        enum Color{
+            RED, BLUE
+        };
+
+        Color side;
+        Player(Color c, string str);
+
+        string getRole()const{ return role;}
+        int makeLaunch()const{ return rand()%(6-1+1)+1;}
+
+};
+
+
+Player::Player(Color c, string str){
+    this->side = c;
+    this->role = str;
+}
+
+
+void results(int* arr){
+    cout << arr[2] <<"(N)\n";
+    cout << arr[1] <<"(M)\n";
+    cout << arr[0] <<"(O)\n";
+}
+
+
+static void match(Player& x, Player& y, int* arr1, int* arr2){
+    switchSort(arr1, 3);
+    switchSort(arr2, 3);
+
+    cout << "Red dices:" << endl;
+    results(arr1);
+    cout << endl;
+    cout << "Blue dices" << endl;
+    results(arr2);
+
+    cout << endl << endl;
+    string s = "OMN";
+
+    for(int i = 2; i >= 0; i--){
+        if(arr1[i] <= arr2[i]){
+            cout << s[i] <<" "<< arr1[i] <<"\tvs\t" << arr2[i] << "   " << "==>   ";
+            cout << "blue wins\n";
+        }
+        else{
+            cout << s[i] <<" "<< arr1[i] <<"\tvs\t" << arr2[i] << "   " << "==>   ";
+            cout << "red wins\n";
+        }
+    }
+}
+
+
+void switchSort(int* arr, int n){
+    for(int i = 0; i < n; i++){
+        for(int j = i+1; j < n; j++){
+            if(arr[i] > arr[j])
+                swap(arr[i], arr[j]);
+        }
+    }
```

Comment:
Personalmente cerco di usare sempre più parentesi graffe, tranne se sviluppo programmi in python, dove parentesi e ; non vengono totalmente usati

Capisco che in C++ si possono scrivere istruzioni singole senza dover utilizzare le parentesi graffe, però cerca di mantenere lo stesso stile quando scrivi codice, o per un'istruzione scrivi le parentesi per tutto, sia per i cicli for sia per quell'if, oppure le usi sempre (consigliato)

Consiglio di usare sempre le parentesi graffe anche per una singola istruzione per evitare problemi di fraintendimenti di codice

Reply:
Ti ringrazio molto per la tua considerazione! Capisco che è più consono mantenere uno stile per rendere più comprensibile il codice per me e per gli altri sviluppatori. Grazie ancora del consigli, mi è molto utile!